#!/usr/bin/env python
PACKAGE = "swanson_base"

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("max_speed", double_t, 0, "Max velocity (m/s) of robot",  1.0, 0,   10.0)
gen.add("base_width", double_t, 0, "Width (m) of robot's wheelbase",  0.42, 0,   5.0)
gen.add("wheel_diameter", double_t, 0, "Width (m) of robot's wheel. [WARNING] Currently has no effect",  0.1905, 0,   5.0)
gen.add("max_turn_radius", double_t, 0, "Maximum turn radius (m). [WARNING] Currently has no effect",  0.381, 0,   5.0)
gen.add("qpps_per_meter", int_t, 0, "Number of encoder pulses per meter",  9596, 0,   20000)
gen.add("flip_cmd_turn_direction",   bool_t,   0, "Flips the commanded direction of the robot",  False)
gen.add("flip_sensed_turn_direction",   bool_t,   0, "Flips direction of the calculated odometry heading",  False)

exit(gen.generate(PACKAGE, "swanson_base", "Drivetrain"))
